### PR TITLE
Feature/GitHub api

### PR DESCRIPTION
**Pull request by DMips**
Pulls latest commit from github and assigns badge if first commit was made by author of github repo who should be mentee.

### CODE REVIEW

File: lib/github_api/fetcher.rb [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    def parse_github_link
+      match = @project.github_link.match(GITHUB_REGEXP)
+      return unless match
+
+      {owner: match[3], repoName: match[4]}
+    end
+
+    def assign_badge
+      unless @project.mentee.badges.coder.any?
+        @project.mentee.badges << Badge.find_by(name: 10)
```
Comment (anjali-sharma) [inline lib/github_api/fetcher.rb]: @DMips Not using the enum ?

Comment (DMips) [inline lib/github_api/fetcher.rb]: @anjali-sharma sure I can change it to Badge.coder

File: lib/github_api/fetcher.rb [hunk 2]
```diff
@@ -37,7 +38,11 @@ def check_latest_commit
 
       unless results.errors.any?
         commits = results.data.repository.ref.target.history.edges
-        assign_badge if commits.length == 1 && commits.first.node.authoredByCommitter
+
+        if commits.length == 1
+          assign_badge if commits.first.node.authoredByCommitter
+          @project.update_column :last_commit, commits.first.node.committedDate
```
Comment (anjali-sharma) [inline lib/github_api/fetcher.rb]: @DMips here the value the `last_commit` has is actually the commit date for the first commit. Does it need to be the latest commit instead?

Comment (DMips) [inline lib/github_api/fetcher.rb]: @anjali-sharma no. history works like git log so first refers to last one which I know isn't confusing at all ;)

Comment (anjali-sharma) [inline lib/github_api/fetcher.rb]: oh 😄  ... didn't know that 👍

**Comment by DMips** [comment]
@RMaksymczuk I'll add tests some time this week.

And I think we have rollbar integration...at least we had it at some point will check on it ;)

**Comment by RMaksymczuk** [comment]
You're right, we have a rollbar configured for the project! :) 

But as I checked on `rollbar` gem documentation (https://github.com/rollbar/rollbar-gem) it works automarically only for controller errors. If we want to get notified for rake tasks, we would need to catch the error and call something like:

`Rollbar.error(e, 'Obtaining Github commits with GraphQL failed')`

**Pull request merged**